### ISSUE TITLE
Mention .well-known based AS discovery mechanism

### ISSUE THREAD

**Issue by dmitrizagidulin** [bspk/oauth.xyz-site#4]
Placeholder issue from our previous conversation -- it would be great to specify a mechanism for an AS to signal to a client that it supports the OAuth.xyz protocol.

Something like:

`GET /.well-known/oauth.xyz`

Response:
```
{
  version: "1.0",
  transaction_endpoint: "https://example.com/transaction"
}
```

(or whatever property names you prefer).

**Comment by tplooker** [comment]
@dmitrizagidulin and I had a good discussion about this today, is this resource not also required for the AS to indicate what it supports in terms of `capabilities` so that the client preparing a request can prepare it accordingly, i.e does the server support jwsd, httpsig, etc?

**Comment by jricher** [comment]
@tplooker I think it's tempting for the server to put a lot of info into a discovery document to allow for programmatic configuration. But, most of the time in the real world, clients don't really auto-configure themselves based on a server's capabilities. Clients (and client software) generally get configured to do one thing by the developer, and just do that. The `capabilities` block is more for the client to send possible options to the server, and for the server to choose from them in its response.

**Comment by tplooker** [comment]
@jricher, fair point, I can see the added complexity for a client if a reliance on the metadata is established in this way. Is it safe to assume that there will be one mandatory or default capability (e.g JWSD, MTLS, HTTPSIG) AS's must support so a client can suitably prove possession of their key and optionally integrity protect their request? Because without a default for this or required metadata describing this from the AS, would a client not be forced into a trial and error type scenario whereby the client repeatedly sends the request, iterating through these capabilities until they found one the server supported?

**Comment by jricher** [comment]
The "mandatory to implement" bit hasn't been settled at all yet, and I agree that there should be at least one method that will always work. I'd even be happy with that being MTLS, since it's much more well-established than the rest.
